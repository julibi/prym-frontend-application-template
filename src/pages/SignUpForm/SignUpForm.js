import React, { useState, useEffect } from 'react';
import { useDispatch } from 'react-redux';
import InputField from '../../components/InputField';
import { useLocalStorage } from '../../hooks/useLocalStorage';
import { SIGNUP } from '../../redux/signUp/actions/signUp.actions';
import style from './SignUpForm.module.scss';

const SignUpForm = () => {
  const storedFormData = JSON.parse(window.localStorage.getItem('formData'));
  const minLengths = {
    password: 6,
    lastname: 2,
    firstname: 2,
    street: 4,
    zip: 5,
    city: 4
  };

  let initialState = {
    lastname: '',
    firstname: '',
    nickname: '',
    email: '',
    password: '',
    pwdConfirmation: '',
    street: '',
    house: '',
    zip: '',
    city: '',
    additionalInfo: ''
  };
  

  const [showAddress, setShowAddress] = useState(false);
  const [formData, setFormData] = useState(storedFormData ? storedFormData : initialState);
  const [formErrors, setFormErrors] = useState(initialState);
  const [isFormDataValid, setIsFormDataValid] = useState(false);
  const [formDataLS, setFormDataLS] = useLocalStorage('formData', initialState);
  const dispatch = useDispatch();
  // https://www.itsolutionstuff.com/post/react-email-validation-exampleexample.html
  const mailPattern = new RegExp(/^(("[\w-\s]+")|([\w-]+(?:\.[\w-]+)*)|("[\w-\s]+")([\w-]+(?:\.[\w-]+)*))(@((?:[\w-]+\.)*\w[\w-]{0,66})\.([a-z]{2,6}(?:\.[a-z]{2})?)$)|(@\[?((25[0-5]\.|2[0-4][0-9]\.|1[0-9]{2}\.|[0-9]{1,2}\.))((25[0-5]|2[0-4][0-9]|1[0-9]{2}|[0-9]{1,2})\.){2}(25[0-5]|2[0-4][0-9]|1[0-9]{2}|[0-9]{1,2})\]?$)/i);
  //https://stackoverflow.com/questions/32534109/javascript-password-validation-at-least-2-number-digits
  const pwPattern = new RegExp('(?=^.{6,}$)(?=(.*\\d){2}).*');
  const toggle = () => {
    setShowAddress(!showAddress)
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    setFormData(initialState);
    setFormDataLS(initialState);
    setFormErrors(initialState)
    dispatch(SIGNUP(formData));
    alert('Form was submitted');
  };

  const handleInput = (event) => {
    const { name, value } = event.target;
    setFormData({ ...formData, [name]: value });
    setFormDataLS({ ...formData, [name]: value });
  };

  const fieldLengthError = (maxLength) => {
    return `This field must contain at least ${maxLength} characters.`
  };

  useEffect(() => {
    let errors = {};
    if (formData.lastname.length < minLengths.lastname) {
      errors = {...errors, lastname: fieldLengthError(minLengths.lastname)};
    }
    if (formData.firstname.length < minLengths.firstname) {
      errors = {...errors, firstname: fieldLengthError(minLengths.firstname)};
    }
    if (!mailPattern.test(formData.email)) {
      errors = {...errors, email: 'This field must contain a valid email address.'};
    }
    if (!pwPattern.test(formData.password)) {
      errors = {...errors, password: 'This password must contain at least 2 numbers.'};
    }
    if (formData.password.length < minLengths.password) {
      errors = {...errors, password: fieldLengthError(minLengths.password)};
    }
    if (formData.password !== formData.pwdConfirmation) {
      errors = {...errors, pwdConfirmation: 'The passwords have to be identical.'};
    }
    if(formData.street.length < minLengths.street) {
      errors = {...errors, street: fieldLengthError(minLengths.street)};
    }
    if(formData.zip.length < minLengths.zip) {
      errors = {...errors, zip: fieldLengthError(minLengths.zip)};
    }
    if(formData.city.length < minLengths.city) {
      errors = {...errors, city: fieldLengthError(minLengths.city)};
    }
    if(formData == initialState) {
      errors = {};
    }
    setFormErrors(errors);
  }, [formData]);

  useEffect(() => {
    const errorExists = !!Object.values(formErrors).filter(error => error.length).length;
    const isRequiredField = (key) => (key !== 'nickname') && (key !== 'additionalInfo') && (key !== 'house');
    const allFieldsFilled = () => {
      for (const key in formData) {
        if (isRequiredField(key)) {
          if (formData[key] == initialState[key]) {
            return false;
          }
        }
      }
      return true;
    }

    if (!errorExists && allFieldsFilled()) {
      setIsFormDataValid(true);
    } else {
      setIsFormDataValid(false);
    }

  }, [formData, formErrors]);

  return(
    <div className={style.signUpWrapper}>
      <fieldset>
        <form onSubmit={(event) => handleSubmit(event)} >
          <InputField
            value={formData.lastname}
            label='Last name'
            placeholder='Mustermann'
            name='lastname'
            onChange={handleInput}
            errorMessage={formErrors.lastname}
          />
          <InputField
            value={formData.firstname}
            label='First name'
            placeholder='Maxime'
            name='firstname'
            onChange={handleInput}
            errorMessage={formErrors.firstname}
          />
          <InputField
            value={formData.nickname}
            label='Nick name'
            placeholder='Maxi'
            name='nickname'
            onChange={handleInput}
          />
          <InputField
            value={formData.email}
            label='Email'
            placeholder='Maxi@gmail.com'
            name='email'
            onChange={handleInput}
            errorMessage={formErrors.email}
          />
          <InputField
            value={formData.password}
            label='Password'
            name='password'
            placeholder='Your smart password'
            type='password'
            onChange={handleInput}
            errorMessage={formErrors.password}
          />
          <InputField
            value={formData.pwdConfirmation}
            label='Repeat password'
            name='pwdConfirmation'
            placeholder='1d!43g@OY05twm'
            type='password'
            onChange={handleInput}
            errorMessage={formErrors.pwdConfirmation}
          />
          <label>{'Show address'}</label>
          <input
            type='checkbox'
            checked={showAddress}
            onChange={toggle}
            className={style.checkbox}
          />
          <br />
          { showAddress && (
            <>
              <InputField
                value={formData.street}
                label='Street'
                placeholder='Mustermann StraÃŸe'
                name='street'
                onChange={handleInput}
                errorMessage={formErrors.street}
              />
              <InputField
                value={formData.house}
                label='House/Apartment'
                placeholder='10'
                name='house'
                onChange={handleInput}
                errorMessage={formErrors.house}
              />
              <InputField
                value={formData.zip}
                label='ZIP'
                placeholder='11233'
                name='zip'
                onChange={handleInput}
                errorMessage={formErrors.zip}
              />
              <InputField
                value={formData.city}
                label='City'
                placeholder='Porto'
                name='city'
                onChange={handleInput}
                errorMessage={formErrors.city}
              />
            </>
          )}
          <label>{'Additional information'}</label>
          <textarea
            className={style.textarea}
            value={formData.additionalInfo}
            name='additionalInfo'
            onChange={handleInput}
            rows={5}
          />
          <input type='submit' value='Submit' className={style.submitButton} disabled={!isFormDataValid && "disabled"}/>
        </form>
      </fieldset>
    </div>
  );
};

export default SignUpForm;