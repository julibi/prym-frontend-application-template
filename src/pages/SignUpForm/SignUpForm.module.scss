.signUpWrapper {
  font-size: large;
  width: 100%;
  max-width: 800px;

  .checkbox {
    margin-bottom: 24px;
  }

  .textarea {
    width: 100%;
    margin: 12px 0 16px 0;
    font-size: large;
  }

  .submitButton {
    height: 40px;
    width: 200px;
    margin-bottom: 16px;
  }
}