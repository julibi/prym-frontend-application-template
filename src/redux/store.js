import { applyMiddleware, combineReducers, createStore } from "redux";
import { composeWithDevTools } from 'redux-devtools-extension';
import { signUpReducer } from "./signUp/reducer";
import { signUpMiddleware } from "./signUp/middlewares";

const middlewares = [signUpMiddleware];

const rootReducer = combineReducers({
  signUp: signUpReducer,
});

const store = createStore(rootReducer, composeWithDevTools(applyMiddleware(...middlewares)));
export { store };
