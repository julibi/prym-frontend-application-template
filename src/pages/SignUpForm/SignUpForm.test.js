  
import React from 'react';
import { render, fireEvent} from '@testing-library/react';
import SignUpForm from './SignUpForm';
import { Provider } from "react-redux";
import { store } from "../../redux/store";

describe('SignUpForm', () => {
  const renderSignUp = () => render(<Provider store={store}><SignUpForm /></Provider>);
  describe('lastname field', () => {
    it('should show error Message, when minimum character amount of field is not reached', () => {
      const { getByPlaceholderText, container } = renderSignUp();
      const input = getByPlaceholderText('Mustermann');
      
      fireEvent.change(input, {
        target: {
          value: 'Y',
        },
      });
  
      expect(container).toMatchSnapshot();
    });
  
    it('should render correctly when conditions are met', () => {
      const { getByPlaceholderText, container } = renderSignUp();
      const input = getByPlaceholderText('Mustermann');
      
      fireEvent.change(input, {
        target: {
          value: 'Yi',
        },
      });
  
      expect(container).toMatchSnapshot();
    });
  });

  describe('email field', () => {
    it('must match valid e-mail pattern', () => {
      const { getByPlaceholderText, container } = renderSignUp();
      const input = getByPlaceholderText('Maxi@gmail.com');
      
      fireEvent.change(input, {
        target: {
          value: '@IamA@fake@mail',
        },
      });

      expect(container).toMatchSnapshot();
    });

    it('renders correctly with valid email', () => {
      const { getByPlaceholderText, container } = renderSignUp();
      const input = getByPlaceholderText('Maxi@gmail.com');
      
      fireEvent.change(input, {
        target: {
          value: 'valid@email.com',
        },
      });

      expect(container).toMatchSnapshot();
    });
  });

  describe('password', () => {
    it('shows error when input is too short (or contains not enough digits)', () => {
      const { getByPlaceholderText, container } = renderSignUp();
      const input = getByPlaceholderText('Your smart password');
      
      fireEvent.change(input, {
        target: {
          value: 'short',
        },
      });

      expect(container).toMatchSnapshot();
    });

    it('renders correctly when at least 6 characters long and container at least 2 digist', () => {
      const { getByPlaceholderText, container } = renderSignUp();
      const input = getByPlaceholderText('1d!43g@OY05twm');
      
      fireEvent.change(input, {
        target: {
          value: 's2hort5!fd6',
        },
      });

      expect(container).toMatchSnapshot();
    });
  });

  describe('submit button', () => {
    // TODO: test the checkbox (snapshot)
    // TODO: write an abstraction function for filling of form
    // TODO: test submit button disablement
    it('only is disabled when all required fields are filled in and valid', () => {
      const { getByPlaceholderText, container } = renderSignUp();
      const lastname = getByPlaceholderText('Mustermann');
      const firstname = getByPlaceholderText('Maxime');
      const email = getByPlaceholderText('Maxi@gmail.com');
      const password = getByPlaceholderText('Your smart password');
      const pwdConfirmation = getByPlaceholderText('1d!43g@OY05twm');

      const street = getByPlaceholderText('Mustermann Stra√üe');
      const zip = getByPlaceholderText('11233');
      const city = getByPlaceholderText('Porto');
      
      fireEvent.change(input, {
        target: {
          value: 'Kent',
        },
      });

      expect(container).toMatchSnapshot();
    });
  });
});