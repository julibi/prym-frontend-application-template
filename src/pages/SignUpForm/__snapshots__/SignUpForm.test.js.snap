// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpForm email field must match valid e-mail pattern 1`] = `
<div>
  <div
    class="signUpWrapper"
  >
    <fieldset>
      <form>
        <label
          class="label"
          for="lastname"
        >
          Last name
        </label>
        <input
          class="inputField"
          name="lastname"
          placeholder="Mustermann"
          type="text"
          value="Yi"
        />
        <p
          class="errorMessage"
        />
        <label
          class="label"
          for="firstname"
        >
          First name
        </label>
        <input
          class="inputField"
          name="firstname"
          placeholder="Maxime"
          type="text"
          value=""
        />
        <p
          class="errorMessage"
        >
          This field must contain at least 2 characters.
        </p>
        <label
          class="label"
          for="nickname"
        >
          Nick name
        </label>
        <input
          class="inputField"
          name="nickname"
          placeholder="Maxi"
          type="text"
          value=""
        />
        <p
          class="errorMessage"
        />
        <label
          class="label"
          for="email"
        >
          Email
        </label>
        <input
          class="inputField"
          name="email"
          placeholder="Maxi@gmail.com"
          type="text"
          value="@IamA@fake@mail"
        />
        <p
          class="errorMessage"
        >
          This field must contain a valid email address.
        </p>
        <label
          class="label"
          for="password"
        >
          Password
        </label>
        <input
          class="inputField"
          name="password"
          placeholder="Your smart password"
          type="password"
          value=""
        />
        <p
          class="errorMessage"
        >
          This field must contain at least 6 characters.
        </p>
        <label
          class="label"
          for="pwdConfirmation"
        >
          Repeat password
        </label>
        <input
          class="inputField"
          name="pwdConfirmation"
          placeholder="1d!43g@OY05twm"
          type="password"
          value=""
        />
        <p
          class="errorMessage"
        />
        <label>
          Show address
        </label>
        <input
          class="checkbox"
          type="checkbox"
        />
        <br />
        <label>
          Additional information
        </label>
        <textarea
          class="textarea"
          name="additionalInfo"
          rows="5"
        />
        <input
          class="submitButton"
          disabled=""
          type="submit"
          value="Submit"
        />
      </form>
    </fieldset>
  </div>
</div>
`;

exports[`SignUpForm email field renders correctly with valid email 1`] = `
<div>
  <div
    class="signUpWrapper"
  >
    <fieldset>
      <form>
        <label
          class="label"
          for="lastname"
        >
          Last name
        </label>
        <input
          class="inputField"
          name="lastname"
          placeholder="Mustermann"
          type="text"
          value="Yi"
        />
        <p
          class="errorMessage"
        />
        <label
          class="label"
          for="firstname"
        >
          First name
        </label>
        <input
          class="inputField"
          name="firstname"
          placeholder="Maxime"
          type="text"
          value=""
        />
        <p
          class="errorMessage"
        >
          This field must contain at least 2 characters.
        </p>
        <label
          class="label"
          for="nickname"
        >
          Nick name
        </label>
        <input
          class="inputField"
          name="nickname"
          placeholder="Maxi"
          type="text"
          value=""
        />
        <p
          class="errorMessage"
        />
        <label
          class="label"
          for="email"
        >
          Email
        </label>
        <input
          class="inputField"
          name="email"
          placeholder="Maxi@gmail.com"
          type="text"
          value="valid@email.com"
        />
        <p
          class="errorMessage"
        />
        <label
          class="label"
          for="password"
        >
          Password
        </label>
        <input
          class="inputField"
          name="password"
          placeholder="Your smart password"
          type="password"
          value=""
        />
        <p
          class="errorMessage"
        >
          This field must contain at least 6 characters.
        </p>
        <label
          class="label"
          for="pwdConfirmation"
        >
          Repeat password
        </label>
        <input
          class="inputField"
          name="pwdConfirmation"
          placeholder="1d!43g@OY05twm"
          type="password"
          value=""
        />
        <p
          class="errorMessage"
        />
        <label>
          Show address
        </label>
        <input
          class="checkbox"
          type="checkbox"
        />
        <br />
        <label>
          Additional information
        </label>
        <textarea
          class="textarea"
          name="additionalInfo"
          rows="5"
        />
        <input
          class="submitButton"
          disabled=""
          type="submit"
          value="Submit"
        />
      </form>
    </fieldset>
  </div>
</div>
`;

exports[`SignUpForm lastname field should render correctly when conditions are met 1`] = `
<div>
  <div
    class="signUpWrapper"
  >
    <fieldset>
      <form>
        <label
          class="label"
          for="lastname"
        >
          Last name
        </label>
        <input
          class="inputField"
          name="lastname"
          placeholder="Mustermann"
          type="text"
          value="Yi"
        />
        <p
          class="errorMessage"
        />
        <label
          class="label"
          for="firstname"
        >
          First name
        </label>
        <input
          class="inputField"
          name="firstname"
          placeholder="Maxime"
          type="text"
          value=""
        />
        <p
          class="errorMessage"
        >
          This field must contain at least 2 characters.
        </p>
        <label
          class="label"
          for="nickname"
        >
          Nick name
        </label>
        <input
          class="inputField"
          name="nickname"
          placeholder="Maxi"
          type="text"
          value=""
        />
        <p
          class="errorMessage"
        />
        <label
          class="label"
          for="email"
        >
          Email
        </label>
        <input
          class="inputField"
          name="email"
          placeholder="Maxi@gmail.com"
          type="text"
          value=""
        />
        <p
          class="errorMessage"
        >
          This field must contain a valid email address.
        </p>
        <label
          class="label"
          for="password"
        >
          Password
        </label>
        <input
          class="inputField"
          name="password"
          placeholder="Your smart password"
          type="password"
          value=""
        />
        <p
          class="errorMessage"
        >
          This field must contain at least 6 characters.
        </p>
        <label
          class="label"
          for="pwdConfirmation"
        >
          Repeat password
        </label>
        <input
          class="inputField"
          name="pwdConfirmation"
          placeholder="1d!43g@OY05twm"
          type="password"
          value=""
        />
        <p
          class="errorMessage"
        />
        <label>
          Show address
        </label>
        <input
          class="checkbox"
          type="checkbox"
        />
        <br />
        <label>
          Additional information
        </label>
        <textarea
          class="textarea"
          name="additionalInfo"
          rows="5"
        />
        <input
          class="submitButton"
          disabled=""
          type="submit"
          value="Submit"
        />
      </form>
    </fieldset>
  </div>
</div>
`;

exports[`SignUpForm lastname field should show error Message, when minimum character amount of field is not reached 1`] = `
<div>
  <div
    class="signUpWrapper"
  >
    <fieldset>
      <form>
        <label
          class="label"
          for="lastname"
        >
          Last name
        </label>
        <input
          class="inputField"
          name="lastname"
          placeholder="Mustermann"
          type="text"
          value="Y"
        />
        <p
          class="errorMessage"
        >
          This field must contain at least 2 characters.
        </p>
        <label
          class="label"
          for="firstname"
        >
          First name
        </label>
        <input
          class="inputField"
          name="firstname"
          placeholder="Maxime"
          type="text"
          value=""
        />
        <p
          class="errorMessage"
        >
          This field must contain at least 2 characters.
        </p>
        <label
          class="label"
          for="nickname"
        >
          Nick name
        </label>
        <input
          class="inputField"
          name="nickname"
          placeholder="Maxi"
          type="text"
          value=""
        />
        <p
          class="errorMessage"
        />
        <label
          class="label"
          for="email"
        >
          Email
        </label>
        <input
          class="inputField"
          name="email"
          placeholder="Maxi@gmail.com"
          type="text"
          value=""
        />
        <p
          class="errorMessage"
        >
          This field must contain a valid email address.
        </p>
        <label
          class="label"
          for="password"
        >
          Password
        </label>
        <input
          class="inputField"
          name="password"
          placeholder="Your smart password"
          type="password"
          value=""
        />
        <p
          class="errorMessage"
        >
          This field must contain at least 6 characters.
        </p>
        <label
          class="label"
          for="pwdConfirmation"
        >
          Repeat password
        </label>
        <input
          class="inputField"
          name="pwdConfirmation"
          placeholder="1d!43g@OY05twm"
          type="password"
          value=""
        />
        <p
          class="errorMessage"
        />
        <label>
          Show address
        </label>
        <input
          class="checkbox"
          type="checkbox"
        />
        <br />
        <label>
          Additional information
        </label>
        <textarea
          class="textarea"
          name="additionalInfo"
          rows="5"
        />
        <input
          class="submitButton"
          disabled=""
          type="submit"
          value="Submit"
        />
      </form>
    </fieldset>
  </div>
</div>
`;

exports[`SignUpForm password renders correctly when at least 6 characters long and container at least 2 digist 1`] = `
<div>
  <div
    class="signUpWrapper"
  >
    <fieldset>
      <form>
        <label
          class="label"
          for="lastname"
        >
          Last name
        </label>
        <input
          class="inputField"
          name="lastname"
          placeholder="Mustermann"
          type="text"
          value="Yi"
        />
        <p
          class="errorMessage"
        />
        <label
          class="label"
          for="firstname"
        >
          First name
        </label>
        <input
          class="inputField"
          name="firstname"
          placeholder="Maxime"
          type="text"
          value=""
        />
        <p
          class="errorMessage"
        >
          This field must contain at least 2 characters.
        </p>
        <label
          class="label"
          for="nickname"
        >
          Nick name
        </label>
        <input
          class="inputField"
          name="nickname"
          placeholder="Maxi"
          type="text"
          value=""
        />
        <p
          class="errorMessage"
        />
        <label
          class="label"
          for="email"
        >
          Email
        </label>
        <input
          class="inputField"
          name="email"
          placeholder="Maxi@gmail.com"
          type="text"
          value="valid@email.com"
        />
        <p
          class="errorMessage"
        />
        <label
          class="label"
          for="password"
        >
          Password
        </label>
        <input
          class="inputField"
          name="password"
          placeholder="Your smart password"
          type="password"
          value="short"
        />
        <p
          class="errorMessage"
        >
          This field must contain at least 6 characters.
        </p>
        <label
          class="label"
          for="pwdConfirmation"
        >
          Repeat password
        </label>
        <input
          class="inputField"
          name="pwdConfirmation"
          placeholder="1d!43g@OY05twm"
          type="password"
          value="s2hort5!fd6"
        />
        <p
          class="errorMessage"
        >
          The passwords have to be identical.
        </p>
        <label>
          Show address
        </label>
        <input
          class="checkbox"
          type="checkbox"
        />
        <br />
        <label>
          Additional information
        </label>
        <textarea
          class="textarea"
          name="additionalInfo"
          rows="5"
        />
        <input
          class="submitButton"
          disabled=""
          type="submit"
          value="Submit"
        />
      </form>
    </fieldset>
  </div>
</div>
`;

exports[`SignUpForm password shows error when input is too short (or contains not enough digits) 1`] = `
<div>
  <div
    class="signUpWrapper"
  >
    <fieldset>
      <form>
        <label
          class="label"
          for="lastname"
        >
          Last name
        </label>
        <input
          class="inputField"
          name="lastname"
          placeholder="Mustermann"
          type="text"
          value="Yi"
        />
        <p
          class="errorMessage"
        />
        <label
          class="label"
          for="firstname"
        >
          First name
        </label>
        <input
          class="inputField"
          name="firstname"
          placeholder="Maxime"
          type="text"
          value=""
        />
        <p
          class="errorMessage"
        >
          This field must contain at least 2 characters.
        </p>
        <label
          class="label"
          for="nickname"
        >
          Nick name
        </label>
        <input
          class="inputField"
          name="nickname"
          placeholder="Maxi"
          type="text"
          value=""
        />
        <p
          class="errorMessage"
        />
        <label
          class="label"
          for="email"
        >
          Email
        </label>
        <input
          class="inputField"
          name="email"
          placeholder="Maxi@gmail.com"
          type="text"
          value="valid@email.com"
        />
        <p
          class="errorMessage"
        />
        <label
          class="label"
          for="password"
        >
          Password
        </label>
        <input
          class="inputField"
          name="password"
          placeholder="Your smart password"
          type="password"
          value="short"
        />
        <p
          class="errorMessage"
        >
          This field must contain at least 6 characters.
        </p>
        <label
          class="label"
          for="pwdConfirmation"
        >
          Repeat password
        </label>
        <input
          class="inputField"
          name="pwdConfirmation"
          placeholder="1d!43g@OY05twm"
          type="password"
          value=""
        />
        <p
          class="errorMessage"
        >
          The passwords have to be identical.
        </p>
        <label>
          Show address
        </label>
        <input
          class="checkbox"
          type="checkbox"
        />
        <br />
        <label>
          Additional information
        </label>
        <textarea
          class="textarea"
          name="additionalInfo"
          rows="5"
        />
        <input
          class="submitButton"
          disabled=""
          type="submit"
          value="Submit"
        />
      </form>
    </fieldset>
  </div>
</div>
`;

exports[`SignUpForm submit button only is disabled when all required fields are filled in and valid 1`] = `
<div>
  <div
    class="signUpWrapper"
  >
    <fieldset>
      <form>
        <label
          class="label"
          for="lastname"
        >
          Last name
        </label>
        <input
          class="inputField"
          name="lastname"
          placeholder="Mustermann"
          type="text"
          value="Kent"
        />
        <p
          class="errorMessage"
        />
        <label
          class="label"
          for="firstname"
        >
          First name
        </label>
        <input
          class="inputField"
          name="firstname"
          placeholder="Maxime"
          type="text"
          value=""
        />
        <p
          class="errorMessage"
        >
          This field must contain at least 2 characters.
        </p>
        <label
          class="label"
          for="nickname"
        >
          Nick name
        </label>
        <input
          class="inputField"
          name="nickname"
          placeholder="Maxi"
          type="text"
          value=""
        />
        <p
          class="errorMessage"
        />
        <label
          class="label"
          for="email"
        >
          Email
        </label>
        <input
          class="inputField"
          name="email"
          placeholder="Maxi@gmail.com"
          type="text"
          value="valid@email.com"
        />
        <p
          class="errorMessage"
        />
        <label
          class="label"
          for="password"
        >
          Password
        </label>
        <input
          class="inputField"
          name="password"
          placeholder="Your smart password"
          type="password"
          value="short"
        />
        <p
          class="errorMessage"
        >
          This field must contain at least 6 characters.
        </p>
        <label
          class="label"
          for="pwdConfirmation"
        >
          Repeat password
        </label>
        <input
          class="inputField"
          name="pwdConfirmation"
          placeholder="1d!43g@OY05twm"
          type="password"
          value="s2hort5!fd6"
        />
        <p
          class="errorMessage"
        >
          The passwords have to be identical.
        </p>
        <label>
          Show address
        </label>
        <input
          class="checkbox"
          type="checkbox"
        />
        <br />
        <label>
          Additional information
        </label>
        <textarea
          class="textarea"
          name="additionalInfo"
          rows="5"
        />
        <input
          class="submitButton"
          disabled=""
          type="submit"
          value="Submit"
        />
      </form>
    </fieldset>
  </div>
</div>
`;
